<launch>
    <include file="$(find smb_gazebo)/launch/smb_gazebo.launch">
        <arg name="world_file" value="$(find smb_highlevel_controller)/worlds/singlePillar.world"/>
        <arg name="laser_enabled" value="true"/>
        <arg name="laser_scan_min_height" value="-0.2"/>
        <arg name="laser_scan_max_height" value="1.0"/>
    </include>

    <node name="smb_highlevel_controller_node"
          pkg="smb_highlevel_controller"
          type="smb_highlevel_controller_node"
          output="screen">
          <rosparam command="load"
                    file="$(find smb_highlevel_controller)/config/config.yaml"/>
    </node>

    <node name="smb_estop_node"
          pkg="smb_highlevel_controller"
          type="smb_estop_node"
          output="screen">
          <rosparam command="load"
                    file="$(find smb_highlevel_controller)/config/config.yaml"/>
    </node>

    <node name="smb_crash_detector_node"
          pkg="smb_highlevel_controller"
          type="smb_crash_detector_node"
          output="screen">
          <rosparam command="load"
                    file="$(find smb_highlevel_controller)/config/config.yaml"/>
    </node>

    <node name="rviz"
          pkg="rviz"
          type="rviz"
          args="-d $(find smb_highlevel_controller)/config/pillar_config.rviz"/>

    <node name="rqt_multiplot" pkg="rqt_multiplot" type="rqt_multiplot" args="--multiplot-config $(find smb_highlevel_controller)/config/imu_multiplot.xml --multiplot-run-all"/>     
</launch>